### PR TITLE
Integrate AuthenticationProvider lib into Provisioning device lib

### PR DESCRIPTION
**Pull request by abhipsaMisra**
The types in AuthenticationProvider are being used only in Microsoft.Azure.Devices.Provisioning.Client, so moving the types in there.

DPS auth -> IoT hub auth:
```c#
        private IAuthenticationMethod CreateAuthenticationMethodFromAuthProvider(
            AuthenticationProvider provisioningAuth,
            string deviceId)
        {
            VerboseTestLogger.WriteLine($"{nameof(CreateAuthenticationMethodFromAuthProvider)}({deviceId})");

            IAuthenticationMethod auth;
            if (provisioningAuth is AuthenticationProviderX509 x509Auth)
            {
                X509Certificate2 cert = x509Auth.GetAuthenticationCertificate();
                auth = new ClientAuthenticationWithX509Certificate(cert, deviceId);
            }
            else if (provisioningAuth is AuthenticationProviderSymmetricKey symmetricKeyAuth)
            {
                auth = new ClientAuthenticationWithSharedAccessKeyRefresh(symmetricKeyAuth.GetPrimaryKey(), deviceId);
            }
            else
            {
                throw new NotSupportedException($"Unknown provisioning auth type.");
            }

            return auth;
        }
```

### CODE REVIEW

File: SDK v2 migration guide.md [hunk 1]
```diff
@@ -348,7 +348,7 @@ Breaking changes:
 
 #### Other notable breaking changes
 
-- Derived `AuthenticationProvider` types no longer require disposal because of the base class; only select derived types will (e.g., `AuthenticationProviderTpmHsm`.)
```
Comment (abhipsaMisra) [inline SDK v2 migration guide.md:351]: We no longer have any type by the name of `AuthenticationProviderTpmHsm`.

File: provisioning/device/src/Authentication/AuthenticationProviderX509.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    public class AuthenticationProviderX509 : AuthenticationProvider
+    {
+        private readonly X509Certificate2 _clientCertificate;
+        private readonly X509Certificate2Collection _certificateChain;
+
+        /// <summary>
+        /// Creates an instance of this class.
+        /// </summary>
+        /// <remarks>
+        /// Ensure that you dispose any supplied <see cref="X509Certificate2"/> after you are done using it to ensure there are no memory leaks.
```
Comment (abhipsaMisra) [inline provisioning/device/src/Authentication/AuthenticationProviderX509.cs:20]: Added a note for certificate disposal.

**Review by patilsnr — APPROVED** [review]

File: provisioning/device/src/Authentication/AuthenticationProvider.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Provisioning.Client
+{
+    /// <summary>
+    /// The authentication provider used by Azure IoT device clients.
```
Comment (drwill-ms) [inline provisioning/device/src/Authentication/AuthenticationProvider.cs:7]:
```suggestion
    /// The authentication provider used by Azure IoT provisioning device client.
```

File: provisioning/device/src/Authentication/AuthenticationProviderX509.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        }
+
+        /// <summary>
+        /// Gets the certificate used for TLS device authentication.
+        /// </summary>
+        /// <returns>The client certificate used during TLS communications.</returns>
+        public X509Certificate2Collection GetAuthenticationCertificateChain()
+        {
+            return _certificateChain;
+        }
```
Comment (drwill-ms) [inline provisioning/device/src/Authentication/AuthenticationProviderX509.cs:60]: Maybe these should just be auto properties instead of field+method?

**Review by drwill-ms — APPROVED** [review]

File: provisioning/device/src/Authentication/AuthenticationProviderX509.cs (hunk 4 above)
Comment (abhipsaMisra) [reply provisioning/device/src/Authentication/AuthenticationProviderX509.cs:60]: sure

**Review by schoims — APPROVED** [review]

**Pull request merged**